Ginger/Ginger/SourceControl/SourceControlConnDetailsPage.xaml.cs

```diff
@@ -59,13 +59,27 @@ private void Bind()
             SourceControlClassTextBox.IsReadOnly = true;
             SourceControlClassTextBox.IsEnabled = false;
 
-            SourceControlURLTextBox.Text = SourceControlIntegration.GetRepositoryURL( App.UserProfile.Solution.SourceControl);
-                SourceControlURLTextBox.IsReadOnly = true;
-                SourceControlURLTextBox.IsEnabled = false;
-
-                App.ObjFieldBinding(SourceControlUserTextBox, TextBox.TextProperty, App.UserProfile.Solution.SourceControl, SourceControlBase.Fields.SourceControlUser);
-                App.ObjFieldBinding(SourceControlPassTextBox, TextBox.TextProperty, App.UserProfile.Solution.SourceControl, SourceControlBase.Fields.SourceControlPass);
+            SourceControlURLTextBox.Text = SourceControlIntegration.GetRepositoryURL(App.UserProfile.Solution.SourceControl);
+            SourceControlURLTextBox.IsReadOnly = true;
+            SourceControlURLTextBox.IsEnabled = false;
 
+            App.ObjFieldBinding(SourceControlUserTextBox, TextBox.TextProperty, App.UserProfile.Solution.SourceControl, SourceControlBase.Fields.SourceControlUser);
+            App.ObjFieldBinding(SourceControlPassTextBox, TextBox.TextProperty, App.UserProfile.Solution.SourceControl, SourceControlBase.Fields.SourceControlPass);
+            if (SourceControlClassTextBox.Text == "GIT")
+            {
+                lblControlConnectionTimeout.Visibility = Visibility.Hidden;
+                txtSourceControlConnectionTimeout.Visibility = Visibility.Hidden;
+            }
+            else
+            {
+                App.ObjFieldBinding(txtSourceControlConnectionTimeout, TextBox.TextProperty, App.UserProfile.Solution.SourceControl, nameof(SourceControlBase.SourceControlTimeout));
+                lblControlConnectionTimeout.Visibility = Visibility.Visible;
+                txtSourceControlConnectionTimeout.Visibility = Visibility.Visible;
+                if (App.UserProfile.SolutionSourceControlTimeout == 80)
```

Comment:
what is this if condition ? If we are binding it then text box will have the value no?